python_bindings/bindings.cpp

```diff
@@ -74,60 +74,97 @@ inline void ParallelFor(size_t start, size_t end, size_t numThreads, Function fn
 
 
 inline void assert_true(bool expr, const std::string & msg) {
-    if (expr == false)
-    throw std::runtime_error("Unpickle Error: "+msg);
+    if (expr == false) throw std::runtime_error("Unpickle Error: " + msg);
     return;
 }
 
 
-template<typename dist_t, typename data_t=float>
-class Index {
-public:
-  Index(const std::string &space_name, const int dim) :
-  space_name(space_name), dim(dim) {
-    normalize=false;
-    if(space_name=="l2") {
-      l2space = new hnswlib::L2Space(dim);
-    }
-    else if(space_name=="ip") {
-      l2space = new hnswlib::InnerProductSpace(dim);
-    }
-    else if(space_name=="cosine") {
-      l2space = new hnswlib::InnerProductSpace(dim);
-      normalize=true;
+inline void set_input_array_shapes(const py::buffer_info& buffer, size_t* rows, size_t* features) {
+    if (buffer.ndim != 2 && buffer.ndim != 1) throw std::runtime_error("data must be a 1d/2d array");
+    if (buffer.ndim == 2) {
+        *rows = buffer.shape[0];
+        *features = buffer.shape[1];
     } else {
-      throw std::runtime_error("Space name must be one of l2, ip, or cosine.");
+        *rows = 1;
+        *features = buffer.shape[0];
     }
-    appr_alg = NULL;
-    ep_added = true;
-    index_inited = false;
-    num_threads_default = std::thread::hardware_concurrency();
+}
 
-    default_ef=10;
-  }
 
-  static const int ser_version = 1; // serialization version
+inline std::vector<size_t> get_input_ids(const py::object& ids_, size_t rows) {
```

Comment:
Seems like this function translates ids_ into std::vector<size_t> ignoring the possible shape. It is not clear from the naming... 
Also not clear why we need `rows` here. There is a check, but I guess it should be done outside of the conversion.

Reply:
Will be addressed in next PR